catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -508,28 +520,135 @@ public void init() {
+          }
+
           catalogFramework.delete(new DeleteRequestImpl(id));
+
           subscriptions.removeSubscriptions(id);
           return ImmutableMap.of("message", "Successfully deleted.");
         },
         util::getJson);
 
+    get(
+        "/workspace/:id/queries",
+        (req, res) -> {
+          String id = req.params(":id");
+          WorkspaceMetacardImpl workspace = getWorkspaceMetacard(id);
+
+          return workspaceService
+              .getQueryMetacards(workspace)
+              .stream()
+              .map(queryTransformer::transformMetacardIntoMap)
+              .collect(Collectors.toList());
+        },
+        util::getJson);
+
+    post(
+        "/query",
+        APPLICATION_JSON,
+        (req, res) -> {
+          Map<String, Object> body = mapper.parser().parseMap(util.safeGetBody(req));
+
+          String workspaceId = (String) body.get("workspaceId");
+          getWorkspaceMetacard(workspaceId);
+
+          Metacard metacard =
+              queryTransformer.transformJsonIntoMetacard((Map<String, Object>) body.get("query"));
+          Metacard saved = saveMetacard(metacard);
+
+          Metacard association =
+              workspaceService.createWorkspaceQueryAssociation(workspaceId, saved.getId());
+          saveMetacard(association);
+
+          return queryTransformer.transformMetacardIntoMap(saved);
+        },
+        util::getJson);
+
+    put(
+        "/query/:id",
+        APPLICATION_JSON,
+        (req, res) -> {
+          String id = req.params("id");
+          Map<String, Object> body = mapper.parser().parseMap(util.safeGetBody(req));
+
+          Metacard metacard = queryTransformer.transformJsonIntoMetacard(body);
+          Metacard updated = updateMetacard(id, metacard);
+
+          return queryTransformer.transformMetacardIntoMap(updated);
+        },
+        util::getJson);
+
+    get(
+        "/query/:id",
+        (req, res) -> {
+          String id = req.params("id");
+          Metacard metacard = util.getMetacard(id);
+
+          if (metacard.getTags().contains(QueryMetacardTypeImpl.QUERY_TAG)) {
+            return queryTransformer.transformMetacardIntoMap(metacard);
+          } else {
+            res.status(400);
+            return ImmutableMap.of("message", "requested ID is not a query");
+          }
+        },
+        util::getJson);
+
+    delete(
+        "/query/:id",
+        APPLICATION_JSON,
+        (req, res) -> {
+          String queryId = req.params(":queryId");
+
+          String[] associationIds =
+              workspaceService
+                  .getAssociationMetacards(queryId)
+                  .stream()
+                  .map(Metacard::getId)
+                  .toArray(String[]::new);
+
+          catalogFramework.delete(new DeleteRequestImpl(associationIds));
+          catalogFramework.delete(new DeleteRequestImpl(queryId));
+
+          return ImmutableMap.of("message", "Successfully deleted.");
```

Comment:
:question: do you not need to set any status codes before returning?

Reply:
It's default status code is `200`. Only if an error occurs should I manually set the status code